.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,52 @@
+#
+# This source file is part of the StanfordBDHG HealtyLLM project
+#
+# SPDX-FileCopyrightText: 2025 Stanford University
+#
+# SPDX-License-Identifier: MIT
+#
+
+name: Static Analysis
+
+on:
+  pull_request:
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: Static-Analysis-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  reuse_action:
+    name: REUSE Compliance Check
+    uses: StanfordBDHG/.github/.github/workflows/reuse.yml@v2
+    permissions:
+      contents: read
+  swiftlint:
+    name: SwiftLint
+    uses: StanfordBDHG/.github/.github/workflows/swiftlint.yml@v2
+    permissions:
+      contents: read
+  periphery:
+    name: Periphery
+    uses: StanfordBDHG/.github/.github/workflows/periphery.yml@v2
+    permissions:
+      contents: read
+    with:
+      runsonlabels: '["macOS", "self-hosted"]'
```

Comment:
@LeonNissen We will need to resolve all the periphery warnings before merging the PR.